### DISCUSSION TITLE
What is a ParticleTile, and what is its relationship to the do_tiling flag and to the tiles of a MultiFab?

### DISCUSSION THREAD

**Discussion by n8xm** [AMReX-Codes/amrex#2497] [Q&A] [answered]
It appears to me that we use [ParticleTile](https://amrex-codes.github.io/amrex/doxygen/structamrex_1_1ParticleTile.html) regardless of whether the `do_tiling` flag is set to `true` in the `inputs` file. E.g. in the Electromagnetic PIC tutorial, we initialize `particle_tile` on [lines 114-115 of EMParticleContainerInit.cpp](https://github.com/AMReX-Codes/amrex-tutorials/blob/0da557baac1736dbf99f4a273ebdba3be25dd6dd/ExampleCodes/Particles/ElectromagneticPIC/Source/EMParticleContainerInit.cpp#L114-L115):

```c++
auto& particles = GetParticles(lev);
auto& particle_tile = particles[std::make_pair(mfi.index(), mfi.LocalTileIndex())];
```

Yet `do_tiling` is never set in the `inputs` file, which means AMReX should use the [default value of False](https://amrex-codes.github.io/amrex/docs_html/Particle.html?highlight=particletile#inputs-parameters). Futhermore, according to the [documentation on constructing ParticleContainers](https://amrex-codes.github.io/amrex/docs_html/Particle.html?highlight=particletile#constructing-particlecontainers):
> If tiling is turned off, then every grid has its own Array-of-Structs and Struct-of-Arrays... When tiling is enabled, then each _tile_ gets its own Struct-of-Arrays and Array-of-Structs instead.

I am now wondering: 
1. Does "tile" in the above sentence refer to a ParticleTile? 
2. My impression is that the `do_tiling` flag influences where particle data is stored, but it does not influence whether we do logical tiling over the grids with `MFIter`, nor does it influence whether we use the ParticleTile class. Is this correct? 
3. Given that we index into `particles` with `std::make_pair(mfi.index(), mfi.LocalTileIndex())`, I am assuming that there is exactly one `ParticleTile` associated with each tile that we iterate over using the `MFIIter` (denoted as the variable `mfi`) in the above code. Is my assumption correct?

**Comment by atmyers** [comment, marked as answer]
This sentence:

> If tiling is turned off, then every grid has its own Array-of-Structs and Struct-of-Arrays... When tiling is enabled, then each tile gets its own Struct-of-Arrays and Array-of-Structs instead.

is wrong, or at least misleading. The underlying data structure always stores some number of `ParticleTile` objects per grid and level. When `do_tiling = 0`, there is always exactly one tile per grid, but the object that gets returned by the `[]` operator of `ParticleLevel is still a `ParticleTile`. When `do_tiling = 1`, each grid can have multiple `ParticleTile` objects, depending on the `tile_size`. Setting `do_tiling = 0` is equivalent to setting the `tile_size` to be very large in each direction.

To answer your specific questions:

1. Yes, but I will change that sentence to make it more clear / correct.
2. Yes.
3. Yes.
